orca-queue/src/main/kotlin/com/netflix/spinnaker/orca/q/QueueShovel.kt

```diff
@@ -71,17 +75,54 @@ class QueueShovel(
     registry.counter(pollOpsRateId).increment()
     previousQueue.poll { message, ack ->
       try {
+        log.debug("Shoveling message $message")
+
+        // transfer the ownership _before_ pushing the message on the queue
+        // we don't want a task handler running that message if the execution is not local
+        transferOwnership(message)
+
         queue.push(message)
         ack.invoke()
         registry.counter(shoveledMessageId).increment()
+      } catch (e: ExecutionNotFoundException) {
+        // no need to log the stack trace on ExecutionNotFoundException, which can be somewhat expected
```

Comment:
why because it might not have been peered yet?

Reply:
yes, that's what I was thinking